### ISSUE TITLE
Alternative to c9.io

### ISSUE THREAD

**Issue by wolffe** [Codiad/Codiad#1071]
With the recent Cloud9 purchase by Amazon, I started looking for hosted alternatives and Codiad is still the first on my list.

Is there an updated fork or is anyone interested in forking it and applying all pull requests and making it PHP 7 compatible?

There are so many fixes and pull requests, but no one seems to manage this repository.

**Comment by chezinut** [comment]
I'm feeling same... dude you can be the one who can will manage it

**Comment by wolffe** [comment]
Anyone interested? Let's vote.

**Comment by guangrei** [comment]
+1

**Comment by cheiff** [comment]
@daeks manages the repository but he is not involved anymore.
I help a bit when I can, but I have no power over pull requests, etc.
I don't know if a fork would be the correct way. Maybe @daeks has any other suggestion on how to proceed?

**Comment by Lord-Y** [comment]
@cheiff is there any way you can contact @daeks to grant you and have more power? Codiad is a great online IDE but there are some bugs that need to be fixed.

**Comment by pwFoo** [comment]
@cheiff @daeks 
Project seems to be no more active maintained?
@CENTRAL-INFO asked to fork codiad.

Alternatives could be: 

- https://github.com/theia-ide/theia (last commit yesterday)
- https://github.com/Fluidbyte/c9-dev-env (last commit 7 months ago...)

Suggestions?

**Comment by Fluidbyte** [comment]
Feel like I should chime in here :)

## The future of Codiad...

I absolutely love Codiad, probably one of the best projects I've worked on, and the community that got involved and kept it alive has been amazing. It was a hobby project for me that I saw turn into something bigger than I ever would have expected.

That being said, I consider the platform to be outdated. The biggest issue for me personally is the lack of testing and DOM-manip nature of jQuery vs. a modern reactive SPA framework. The architecture works at the level where Codiad resides but extending the features and functionality of the app would be a stretch and regressions would become hard to mitigate, so I generally feel, from an architectural standpoint, its future is limited.

As mentioned; it is MIT licensed, so I welcome public forks and any additional code against the application, but I presume maintenance by folks on the "core" will continue to dwindle.

## For anyone looking for alternatives...

If anyone is interested, I would highly suggest [c9/core](https://github.com/c9/core), the public repo is easy to stand up, and I run it similarly to how I ran Codiad; on a DigitalOcean Droplet with fairly minimal resources.

I've maintained a strong interest in web-based editors, and [CodeAnywhere](https://codeanywhere.com), [ShiftEdit](https://shiftedit.com), [Theia](https://github.com/theia-ide/theia), [CodeTasty](https://codetasty.com/) all feel to me like they try to approach things in a similar manner to Codiad - more editor than IDE.

Eclipse Che (project that runs [Koding](https://koding.com) platform) is a much larger option for those looking for a lot more tooling and IDE-like features.

**Comment by Creat1veM1nd** [comment]
Not everyone is able to run the mentioned alternatives on a managed server (no rights to install anything or not having docker). Codiad was perfect to edit files on the server, since one could install it like e.g. a blog. 

I searched quite a while for getting something like Codiad years ago, a tool which helps me to edit files on my managed server while on the go and with things like WinSCP not possible. 

Does anyone have any other alternatives which can achieve what Codiad did ?

**Comment by meepiquitous** [comment]
>Does anyone have any other alternatives which can achieve what Codiad did ?

Yes, kind of - both [elFinder](https://github.com/Studio-42/elFinder) and [kodExplorer](https://github.com/kalcaddle/KodExplorer) are file managers with [ACE Editor](https://github.com/ajaxorg/ace) baked in. You can add other code editors like Microsoft's [Monaco](https://microsoft.github.io/monaco-editor/index.html), but that may be a fair amount of work.

You may also look at [ICEcoder](https://github.com/icecoder/ICEcoder), which isn't nearly as polished.

If you're fine with portable versions of desktop apps, consider [MobaXterm](https://mobaxterm.mobatek.net/), [atom](https://github.com/atom/atom) or [Visual Studio Code](https://github.com/Microsoft/vscode).

Regarding the state of Codiad:

   (@4oo4 on April 1st, 2018; [issue](https://github.com/Andr3as/Codiad-CodeGit/issues/62#issuecomment-377806623))

> I'm closing this since Codiad itself is unmaintained and has at least 4 (!) [open](https://github.com/Codiad/Codiad/issues/1022) [issues](https://github.com/Codiad/Codiad/issues/1021) [for](https://github.com/Codiad/Codiad/issues/1021) [vulnerabilities](https://github.com/Codiad/Codiad/issues/1020), including two potential RCE, which no one has appeared to even followed up on. If those will never get patched in Codiad itself, there's no point in worrying about the security of this plugin.

There's also issue #1017, which at this point has been open for more than a year. 

Which is especially a shame because of the work people have put into addons & themes:

[Clear Theme](https://github.com/daeks/Codiad-Theme-Clear)
<img src="https://raw.githubusercontent.com/daeks/Codiad-Theme-Clear/master/screen.png" width="800" />

[Feisty Theme](https://github.com/tobx/Codiad-Theme-Feisty)
<img src="https://raw.githubusercontent.com/tobx/Codiad-Theme-Feisty/master/screen.png" width="800" />

[Obsidian Theme](https://github.com/sovrin/Codiad-Theme-Obsidian)
<img src="https://raw.githubusercontent.com/sovrin/Codiad-Theme-Obsidian/master/screen.png" width="800" />

[sLite Theme](https://github.com/bigsuraz/sLite)
<img src="https://raw.githubusercontent.com/bigsuraz/sLite/tests/codiad.png" width="800" />


(The reason i mention them here is that the theme installer never worked for me, so i assume most people didn't bother.)

**Comment by 4oo4** [comment]
@Fluidbyte It would be a good idea to update `README.md` to tell people that this isn't maintained anymore, so that anyone that stumbles on this project has accurate expectations of quality/security.

**Comment by Fluidbyte** [comment]
Good call, I've updated the [project readme](https://github.com/Codiad/Codiad/blob/master/README.md#unmaintained-status)

**Comment by unix4you2** [comment]
Hi everyone.
I wrote this online editor some years ago.  Its bootstrap 3 and responsive based in ACE editor.
Maybe one of you want to see it and help to develop it.   This is the main code editor used in Practico framework and has a lot of features:  https://github.com/unix4you2/pcoder
I came here because I was looking for other similar projects to improve mine.  Its a shame that codiad developer stopped it.

If some of you want to continue with codiad I suggest to fork it and build community around the new project.

Regards.

**Comment by zedauni** [comment]
> With the recent Cloud9 purchase by Amazon, I started looking for hosted alternatives and Codiad is still the first on my list.
> 
> Is there an updated fork or is anyone interested in forking it and applying all pull requests and making it PHP 7 compatible?
> 
> There are so many fixes and pull requests, but no one seems to manage this repository.

There is a an updated and currently maintained fork of Codiad which is quite interesting: https://github.com/Atheos/Atheos.

**Comment by zedauni** [comment]
> Not everyone is able to run the mentioned alternatives on a managed server (no rights to install anything or not having docker). Codiad was perfect to edit files on the server, since one could install it like e.g. a blog.
> 
> I searched quite a while for getting something like Codiad years ago, a tool which helps me to edit files on my managed server while on the go and with things like WinSCP not possible.
> 
> Does anyone have any other alternatives which can achieve what Codiad did ?

Give a try to [Atheos](https://github.com/Atheos/Atheos).

**Comment by Creat1veM1nd** [comment]
@zedauni 
I did that and so far I am very happy with it. Thanks for the hint!

**Comment by pwFoo** [comment]
At the moment I use Theia, abut I'll take a look at Atheos. Thanks!
